features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/internal/MiscUtils.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.opentelemetry.internal
+
+import android.os.Build
+import com.datadog.android.api.InternalLogger
+
+internal const val NEEDS_DESUGARING_ERROR_MESSAGE =
+    "Trying to use OpenTelemetry SDK support for Android 23 and below. " +
+        "In order for this to properly work you will need to enable coreDesugaring " +
+        "in your compileOptions"
+
+internal fun <T : Any?> executeSafelyOnAndroid23AndBelow(
+    action: () -> T,
+    default: T,
+    internalLogger: InternalLogger? = null
+): T {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        return action()
+    } else {
+        return try {
+            // we are forcing here the checkup as it could be triggered only at runtime and will be too late
+            Class.forName("java.util.function.Function")
+            action()
+        } catch (@Suppress("TooGenericExceptionCaught") e: Exception) {
```

Comment:
if this code block will be executed in the hot path, we need an optimization to use reflection only once

Reply:
yes, I thought about it myself, this code will only be called in the `OtelTracerProvider` builder method which normally should be happening few times in the clients code when an `OtelTracerProvider` is created. I don't think we should not anything else here.